api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -564,14 +569,26 @@ private static String sanitizeDate(int days, int today) {
     return "(date)";
   }
 
-  private static String sanitizeTimestamp(long micros, long now) {
-    String isPast = now > micros ? "ago" : "from-now";
-    long diff = Math.abs(now - micros);
-    if (diff < FIVE_MINUTES_IN_MICROS) {
+  private static String sanitizeTimestamp(ChronoUnit unit, long timeUnits, long nowMillis) {
+    long timeMillis;
+    switch (unit) {
+      case MICROS:
+        timeMillis = DateTimeUtil.microsToMillis(timeUnits);
+        break;
+      case NANOS:
+        timeMillis = DateTimeUtil.nanosToMillis(timeUnits);
```

Comment:
I don't understand why there are so many changes here. Couldn't this leave `now` in micros and just convert nanos to micros here?

Reply:
I don't see how. See line 521 of `sanitize` for example.

Reply:
@epgif, why does the addition of nanosecond precision require switching `now` throughout this file from microseconds to nanoseconds? It seems to me that this could remain as it was.

Reply:
I see what you mean now. How does it look now?